main/src/ca/uwaterloo/flix/language/ast/MonoType.scala

```diff
@@ -58,7 +58,9 @@ object MonoType {
 
   case class Array(tpe: MonoType) extends MonoType
 
-  case class Channel(tpe: MonoType) extends MonoType
+  case class Sender(tpe: MonoType) extends MonoType
+
+  case class Receiver(tpe: MonoType) extends MonoType
```

Comment:
I think- but I am not sure- due to Lowering we can actually remove both Sender and Receiver from MonoType! (And thus simplify the later phases even more).

Reply:
I got 99% of the way through a version where we had no `Sender` or `Receiver` at all (just using Mmpc and aliases within `Channel.flix`) but that failed because it broke all of our LibNix and LibMin tests.

So then I tried just plumbing `Sender` and `Receiver` through to lowering, but that resulted in incomplete matches. In the end this was the easiest way I could find to make things work, but if you can think of a way of rearranging things which would allow it to work, let me know.

Reply:
I think I can explain this. 

After `Lowering` we should have eliminated all occurrences of `Channel` and instead the type should now be `Channel.Mpmc` (or whatever it is called). 

However, as I point to below, there is still one place where we retain the `Channel` type instead of translating it.

Reply:
... And I was wrong...

If we use `mkChannelTpe` everywhere, how can there still be a `Channel` left over later?